### DISCUSSION TITLE
any idea? rancher stable version not compliant on kube 1.25+

### DISCUSSION THREAD

**Discussion by Gilaco** [kube-hetzner/terraform-hcloud-kube-hetzner#376] [Q&A] [answered]
Hey, 
I wondered if any of you encountered the issue of deploying rancher successfully
I keep on getting some issues due to kube version incompatibility.... seems like rancher works mostly with Kube<1.25

any idea? alternative?

thanks
G

**Comment by mysticaltech** [comment, marked as answer]
@Gilaco Yes, most probably, rancher products tend to lag at few weeks, and since 1.25 k3s just came out, all you have to do for now is set initial_k3s_channel = 'v1.24' in your kube.tf and apply again, that should fix it! And in a few weeks or a month you can move to "v1.25". 

We will update the docs to inform that it is always better to stick to the latest kube - 1, products like AWS EKS are still stuck at 1.22 for instance, so v1.24 is already amazing!

**Reply by Gilaco** [reply]
Thanks! Yeap 1.24 is great - don't know why I didn't think about this option 😂😂

Great project btw! Just taking my initial steps with it and already loving it :-) 

Cheers
G

**Reply by mysticaltech** [reply]
Awesome!! Good to hear.. 🙏

**Reply by otavio** [reply]
And how would you upgrade k3s version without cluster recreation?

**Reply by mysticaltech** [reply]
@otavio This project has the system upgrade controller running by default, so if you did not realize it yet, your cluster is automatically upgrading to the latest version of k3s as soon as it comes out, BUT you can set the update channel.

And all of this magic DOES NOT NEED CLUSTER recreation, it does this node by node, drains, updates k3s, restarts, rejoins the cluster. And no one notices it! Pretty magical stuff these rancher folks accomplished!

**Reply by mysticaltech** [reply]
There are CRDs for the system upgrade controller, you can expect those, and tweak the jobs to your liking.

**Reply by mysticaltech** [reply]
Even if you set the update channel as "v1.24" you still get the updates but for that kube version... So 1.24.x etc.

**Reply by mysticaltech** [reply]
We now also give you an option to disable k3s updates (see kube.tf.example) but IMHO, it's not needed nor wanted, and could be a security risk to not always stay with the latest or greatest.

**Reply by otavio** [reply]
It would be nice to collect this kinda of example in a Wiki or similar, so people have a place to look as reference.

**Reply by mysticaltech** [reply]
Indeed... The thought has crossed my mind. Will do in time!  🙏

**Comment by loxy** [comment]
So, that means that once [v1.25.3+k3s1](https://github.com/k3s-io/k3s/releases/tag/v1.25.3%2Bk3s1) was out, my cluster automatically switched to that Kubernetes version (while specifying stable)?

And if I set initial_k3s_channel = "v1.24" and apply again it will be reverted? With no drawbacks?

I'm asking because I have problems with postgres operator (PGO). It is not compatible with 1.25...
